types/vue.d.ts

```diff
@@ -0,0 +1,49 @@
+// vue augmentation
+import Vue, { ComponentOptions } from 'vue'
+
+import { ValidationRule } from 'vuelidate/lib/validators'
+import { Validation } from "./vuelidate"
+
+interface DynamicValidation<V extends Vue> { }
+
+type ValidationProperties<V extends Vue> = {
+    [P in keyof V]?: Validation
+}
+
+interface ValidationGroups<V extends Vue> {
+    [groupName: string]: ValidationProperties<V>
+}
+
+declare module 'vue/types/vue' {
+    interface Vue {
+        $v: ValidationProperties<this> & ValidationGroups<this> & Validation
+
+        delayTouch(v: Validation): void
+
+        validations(): DynamicValidation<this>
+    }
+}
+
+interface ValidGroupDecl {
+    [group: string]: string[]
+}
+
+
+interface ValidPropertyDecl {
+    [prop: string]: RuleDecl
+}
+
+type ValidationDecl = ValidationRule | ((...args: any[]) => ValidationRule)
+type GroupDecl = string[]
+type AsyncDecl = (...args: any[]) => boolean | Promise<boolean>
+type NestedDecl = RuleDecl
+interface RuleDecl {
+    [rule: string]: ValidationDecl | GroupDecl | AsyncDecl | NestedDecl
+}
+
+
+declare module 'vue/types/options' {
+    interface ComponentOptions<V extends Vue> {
+        validations?: RuleDecl
```

Comment:
With this typing i can't use [Dynamic validation schema](https://monterail.github.io/vuelidate/#sub-dynamic-validation-schema). Maybe its better to write `RuleDecl | (() => RuleDecl)`?

Reply:
hi @xSorc ,

i have this working example: https://github.com/janesser/vue-playground/blob/4ff484eb9f20adc832b520539124afad1c751cba/src/components/valid/valid.ts#L118

what's different about yours?

Reply:
Hi @janesser ,

Is your example realy works? You have duplicate **validations** variable. One in the component decorator and the same one in the list of functions below. And witch one is correct? I'd reproduced this case localy - function **validations** doesn't call.

I think the function **validations** doesn't calls, because you need to declare **validations** variable directly in the root vue object. But vue-property-decorator moves all you functions into the methods section. The right place for function **validations** - in the decorator.

p.s. sorry for bad grammar :)

Reply:
ill have a closer look, thanks for the hints.